auth/auth.go

```diff
@@ -45,19 +46,21 @@ func (user *UserIdentity) Restore(ctx *format.RestoreCtx) error {
 	return nil
 }
 
-// String converts UserIdentity to the format user@host.
+func EscapeAccountName(s string) string {
+	return "'" + strings.ReplaceAll(s, "'", "\\'") + "'"
```

Comment:
```suggestion
	return "`" + strings.ReplaceAll(s, "`", "``") + "`"
```

Doubling the quotation mark is the ISO standard way of escaping. `\` is prone to NO_BACKSLASH_ESCAPES.

But speaking of backslash escapes, if the user name contains a `\` this ReplaceAll is also wrong when @@sql_mode has no NO_BACKSLASH_ESCAPES. So it's the safest just to use the identifier form instead of the string form.

Reply:
Thanks for pointing this out, I missed that!

What I tried to reflect was the error message from MySQL, when implementing `RENAME USER`:
```sql
mysql> rename user "User's"@"root's friend" to u1@localhost;
ERROR 1396 (HY000): Operation RENAME USER failed for 'User\'s'@'root\'s friend'
mysql> select @@sql_mode;
+-----------------------------------------------------------------------------------------------------------------------+
| @@sql_mode                                                                                                            |
+-----------------------------------------------------------------------------------------------------------------------+
| ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION |
+-----------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> set @@sql_mode="ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION,NO_BACKSLASH_ESCAPES"
    -> ;
Query OK, 0 rows affected (0.01 sec)

mysql> select @@sql_mode;
+--------------------------------------------------------------------------------------------------------------------------------------------+
| @@sql_mode                                                                                                                                 |
+--------------------------------------------------------------------------------------------------------------------------------------------+
| ONLY_FULL_GROUP_BY,NO_BACKSLASH_ESCAPES,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION |
+--------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> rename user "User's"@"root's friend" to u1@localhost;
ERROR 1396 (HY000): Operation RENAME USER failed for 'User''s'@'root''s friend'
mysql> 
```
So to be error string compatible, we should probably still use (') "single quote" character, instead of (`) "backtick" character.

I will see if I can make it adjust according to NO_BACKSLASH_ESCAPES.

Do you think we should use (`) backtick and always do double backtick instead of escaping with (\) backslash?

Reply:
Are there anyway to get the context/parser/lexer/sql_mode without changing the types/api for this?
Otherwise I cannot see how to adapt to the sessions sql_mode setting, which would support your suggestion to just use double quoting.